src/gui/widgets/LcdFloatSpinBox.cpp

```diff
@@ -0,0 +1,255 @@
+	m_fractionDisplay(numFrac, style, parent, name, true),
+	m_mouseMoving(false),
+	m_intStep(false),
+	m_origMousePos(),
+	m_displayOffset(0)
+{
+	layoutSetup(style);
+}
+
+
+void LcdFloatSpinBox::layoutSetup(const QString &style)
+{
+	// Assemble the LCD parts
+	QHBoxLayout *lcdLayout = new QHBoxLayout();
+
+	m_wholeDisplay.setSeamless(false, true);
+	m_fractionDisplay.setSeamless(true, false);
+
+	lcdLayout->addWidget(&m_wholeDisplay);  
+
+	QLabel *dotLabel = new QLabel("", this);
+	QPixmap *dotPixmap = new QPixmap(embed::getIconPixmap(QString("lcd_" + style + "_dot").toUtf8().constData()));
+	dotLabel->setPixmap(dotPixmap->copy(0, 0, dotPixmap->size().width(), dotPixmap->size().height() / 2));
+	lcdLayout->addWidget(dotLabel);
+
+	lcdLayout->addWidget(&m_fractionDisplay);
+
+	lcdLayout->setContentsMargins(0, 0, 0, 0);
+	lcdLayout->setSpacing(0);
+
+	// Add space for label
+	QVBoxLayout *outerLayout = new QVBoxLayout();
+	outerLayout->addLayout(lcdLayout);
+	outerLayout->addSpacing(9);
+	outerLayout->setContentsMargins(0, 0, 0, 0);
+	outerLayout->setSizeConstraint(QLayout::SetFixedSize);
+	this->setLayout(outerLayout);
+}
+
+
+void LcdFloatSpinBox::update()
+{
+	const int whole = static_cast<int>(model()->value());
+	const float fraction = model()->value() - static_cast<int>(model()->value());
+	const int intFraction = fraction * std::pow(10.f, m_fractionDisplay.numDigits());
+	m_wholeDisplay.setValue(whole + m_displayOffset);
+	m_fractionDisplay.setValue(intFraction + m_displayOffset);
+
+	QWidget::update();
+}
+
+
+void LcdFloatSpinBox::contextMenuEvent(QContextMenuEvent* event)
+{
+	// for the case, the user clicked right while pressing left mouse-
+	// button, the context-menu appears while mouse-cursor is still hidden
+	// and it isn't shown again until user does something which causes
+	// an QApplication::restoreOverrideCursor()-call...
+	mouseReleaseEvent(nullptr);
+
+	CaptionMenu contextMenu(model()->displayName());
+	addDefaultActions(&contextMenu);
+	contextMenu.exec(QCursor::pos());
+}
+
+
+void LcdFloatSpinBox::mousePressEvent(QMouseEvent* event)
+{
+	if (event->button() == Qt::LeftButton &&
+		!(event->modifiers() & Qt::ControlModifier) &&
+		event->y() < m_wholeDisplay.cellHeight() + 2)
+	{
+		m_mouseMoving = true;
+		m_origMousePos = event->globalPos();
+		QApplication::setOverrideCursor(Qt::BlankCursor);
+
+		AutomatableModel *thisModel = model();
+		if (thisModel)
+		{
+			thisModel->addJournalCheckPoint();
+			thisModel->saveJournallingState(false);
+		}
+	}
+	else
+	{
+		FloatModelView::mousePressEvent(event);
+	}
+}
+
+
+void LcdFloatSpinBox::mouseMoveEvent(QMouseEvent* event)
+{
+	// switch between integer and fractional step based on cursor position
+	if (event->x() < m_wholeDisplay.width()) {m_intStep = true;}
+	else {m_intStep = false;}
+
+	if (m_mouseMoving)
+	{
+		int dy = event->globalY() - m_origMousePos.y();
+		if (gui->mainWindow()->isShiftPressed())
+			dy = qBound(-4, dy/4, 4);
```

Comment:
```suggestion
		if (gui->mainWindow()->isShiftPressed()) { dy = qBound(-4, dy/4, 4); }
```

Even if multiline `if` is preferred, explicit blocking must be used.